### ISSUE TITLE
Is library still maintained ? Are PRs accepted ?

### ISSUE THREAD

**Issue by olup** [Paratron/hookrouter#112]
Pretty much just the title ;-)

**Comment by Paratron** [comment]
Yeah, I mentioned in the past that I have extremely low free time to work on it. But I _am_ working on version 2.0 of the router.

PRs are meaningless right now, since I do a complete rewrite of the codebase. Since we are now in lockdown because of Covid19 here in germany, I hopefully get some more time to work on it, if my daughters sleep well at night. Yesterday I made good progress in writing the router core code and will try to add react and preact connectors tonight. 

Stay tuned - its not dead :)

**Comment by olup** [comment]
Good to hear, this lib has great potential. Are you writing it up again in
Typescript ? Would be good to have types natively embedded.

**Comment by olup** [comment]
Sorry this came wrong from my inbox. Was wondering : are you rewriting this in typescript by any chance? We would greatly benefit from proper types.

**Comment by Paratron** [comment]
I am rewriting this, since it was initially only a proof of concept and suffered from a couple of architectual problems.

I will not write any library ever in typescript. I dislike typescript for a number of reasons and will never use it if I don't absolutely have to.

**Comment by StallionV** [comment]
Proof of concept ?
(Is shocked and passes out)

**Comment by Paratron** [comment]
I think I did mention it in my initial article about it. I wrote down the idea of it in the matter of two hours during a weekend and it was suprisingly stable but in the long run turned out to have some issues that made it hard to implement serverside rendering.

**Comment by JasonAtallah** [comment]
@Paratron will version 2 have a lot of breaking changes? If so, will there be a feasible migration path? Also, is using the current version in production discouraged?

**Comment by kyeotic** [comment]
I was impressed at the size and simplicity of the hookrouter API when I first tried it. When development stopped and bugs turned into blockers I decided to build my hook-based router.

For those of you like @olup who are looking for a path forward [raviger](https://github.com/kyeotic/raviger) just released its [v2](https://github.com/kyeotic/raviger/releases/tag/v2.0.0). It has been under active development since its start two years ago, is in use in several production systems, and most bugs are closed within a day or two. The API is not drop-in compatible, but its close enough that it should be easy for most people to switch.

**Comment by olup** [comment]
Well, I believe react-router v6 now supports very similar DX, so I think it remains the better choice.

**Comment by kyeotic** [comment]
@olup That probably true if DX or familiarity are your top concerns. React-router very large compared to raviger or hookrouter, more than 5x the size gzipped. If bundle size is a high priority you might consider a slimmer option :)
